components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -260,74 +272,60 @@ abstract class AbstractFetchDownloadService : Service() {
             */
             val uiStatus = getDownloadJobStatus(download)
 
-            // Dispatch the corresponding notification based on the current status
-            val notification = when (uiStatus) {
-                DownloadJobStatus.ACTIVE -> {
-                    DownloadNotification.createOngoingDownloadNotification(
-                        context,
-                        download
-                    )
-                }
-
-                DownloadJobStatus.PAUSED -> {
-                    DownloadNotification.createPausedDownloadNotification(context, download)
-                }
-
-                DownloadJobStatus.COMPLETED -> {
-                    DownloadNotification.createDownloadCompletedNotification(context, download)
-                }
+            updateForegroundNotificationIfNeeded(download)
 
-                DownloadJobStatus.FAILED -> {
-                    DownloadNotification.createDownloadFailedNotification(context, download)
-                }
+            // Dispatch the corresponding notification based on the current status
+            updateDownloadNotification(uiStatus, download)
 
-                DownloadJobStatus.CANCELLED -> {
-                    NotificationManagerCompat.from(context).cancel(download.foregroundServiceId)
-                    download.lastNotificationUpdate = System.currentTimeMillis()
-                    null
-                }
+            if (uiStatus != ACTIVE) {
+                sendDownloadStopped(download)
             }
+        }
+    }
 
-            notification?.let {
-                NotificationManagerCompat.from(context).notify(
-                    download.foregroundServiceId,
-                    it
-                )
+    /***
```

Comment:
nit: Tripple star 🙃

Reply:
Fixing it 😄 
![image](https://user-images.githubusercontent.com/773158/82920941-ee6e7e80-9f45-11ea-9e1d-8f1de1afcb50.png)